### ISSUE TITLE
Field caps api incorrect reports a doc values only scaled float field as not searchable

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#97394]
Reproduction:

```
PUT /index1
{
    "settings": {
        "index.mode": "time_series",
        "time_series": {
                "start_time": "2000-01-01T00:00:00Z",
                "end_time": "2099-12-31T23:59:59Z"
            },
            "routing_path": [
                "field1"
            ]
    },
    "mappings": {
        "properties": {
            "field1": {
                "type": "keyword",
                "time_series_dimension": true
            },
            "field2": {
                "type": "long",
                "time_series_metric": "gauge"
            },
            "field3": {
                "type": "scaled_float",
                "time_series_metric": "gauge",
                "scaling_factor": 100
            }
        }
    }
}

GET /index1/_field_caps/?fields=field*
```

Field caps response:

```
{
    "indices": [
        "index1"
    ],
    "fields": {
        "field1": {
            "keyword": {
                "type": "keyword",
                "metadata_field": false,
                "searchable": true,
                "aggregatable": true,
                "time_series_dimension": true
            }
        },
        "field3": {
            "scaled_float": {
                "type": "scaled_float",
                "metadata_field": false,
                "searchable": false,
                "aggregatable": true,
                "time_series_metric": "gauge"
            }
        },
        "field2": {
            "long": {
                "type": "long",
                "metadata_field": false,
                "searchable": true,
                "aggregatable": true,
                "time_series_metric": "gauge"
            }
        }
    }
}
```

Both fields aren't indexed by default (because marked as time series metric), but do have doc values.
Which means the fields are searchable, but slow. However for the scaled float field, field caps reports that it isn't searchable. Which isn't correct. Field caps should report that field3 is searchable.

Looking at the mapping code the `ScaledFloatFieldType` extends from `MappedFieldType` and that returns searchable only for indexed fields. Either `MappedFieldType` should overwrite the `isSearchable()` method or extend from `NumberFieldType`.

**Comment by lalit-satapathy** [comment]
Is this fixed as part of 8.10?

**Comment by martijnvg** [comment]
Yes, that is fixed in 8.10. The PR got merged before the 8.10 branch was cut.

**Comment by martijnvg** [comment]
Fixed via https://github.com/elastic/elasticsearch/pull/97724
